### DISCUSSION TITLE
Organization default: "Automatically delete head branches" for new repositories

### DISCUSSION THREAD

**Discussion by DanielButAtWork** [community/community#145139] [Repositories]
### Select Topic Area

General

### Body

In our organization, we have enabled the "Automatically delete head branches" option[1] on existing repositories. However, we sometimes forget to enable this for new repositories. I have searched for documentation on enabling this by default for an organization, or otherwise applying it automatically, and come up empty. The closest I could find was a discussion (#24768) where it was mentioned that there is an API that can be used to configure this setting, so we could create a bot to periodically enable it everywhere, but that seems like a less-than-ideal approach.

Is there a way for an organization to be set up so that all future repositories that are created have "Automatically delete head branches" enable? If yes, this is a request for documentation on how to do that. If not, this is a request to add such a feature.

[1] https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-the-automatic-deletion-of-branches

**Comment by ontramatt** [comment]
Upvote for this, would be an excellent org-wide default setting.

**Reply by salvisolamartinell** [reply]
Please upvote on the original post, not on this comment.

**Comment by waldnzwrld** [comment]
I would also like to see this, It would be extremely useful as I build out new projects.

**Comment by vanngoh** [comment]
It feels like this feature is totally underrated. 
If you were led here for any reason, please show some love by upvoting the original post! ❤️

**Comment by thislooksfun** [comment]
I want this on a personal level as well. I want this on for all my repos, always. Having to enable it every time is really frustrating, and I keep forgetting to turn it on for new repos until after a PR is merged.

**Comment by vedmaka** [comment]
Upvote

**Comment by robmcgee-isw** [comment]
+1

**Comment by theoclipse** [comment]
+1

**Comment by axelboursin** [comment]
+1

We are in a `+75 repositories organization` and this kind of org-setting level can be game changer.

**Comment by georgeorwell05** [comment]
An organization-wide default would definitely help keep repository settings consistent and reduce manual setup for every new project. Small automation features like this can save a surprising amount of time, much like [ZS999 Game](https://techrealm.pk/cz777-game/).

**Comment by hanrycoupin2** [comment]
Good feature request automating this for new repositories would save teams a lot of hassle. [Y 888](https://y888gameapp.com.pk/) is worth exploring too.

**Comment by johnruliana1** [comment]
A useful organization-level default would save a lot of repetitive setup; [2jbet](https://2jbet-game.pk/) is worth checking out too.
